app/src/main/java/com/practicum/playlistmaker/ui/search/fragment/SearchFragment.kt

```diff
@@ -134,7 +140,6 @@ class SearchActivity : AppCompatActivity() {
         viewModel.state.observe(this) {
```

Comment:
🍏 Студия сама подсказывает, что здесь лучше использовать viewLifecycleOwner, так как жизненный цикл фрагмента больше, чем жизненный цикл содержащейся в нем view.

```suggestion
        viewModel.state.observe(viewLifecycleOwner) {
```